opencog/atoms/base/ClassServer.cc

```diff
@@ -45,34 +45,23 @@ ClassServer::ClassServer(const NameServer & nameServer):
 	_nameServer(nameServer)
 {}
 
-static Handle validating_factory(const Handle& atom_to_check)
-{
-	ClassServer::Validator* checker =
-		classserver().getValidator(atom_to_check->get_type());
-
-	/* Well, is it OK, or not? */
-	if (not checker(atom_to_check))
-		throw SyntaxException(TRACE_INFO,
-		     "Invalid Atom syntax: %s",
-		     atom_to_check->to_string().c_str());
-
-	return atom_to_check;
-}
-
-void ClassServer::spliceFactory(Type t, AtomFactory* fact)
+template<typename T>
+void ClassServer::splice(std::vector<T>& methods, Type t, T fact)
```

Comment:
I suggest adding some comments about what this method does, what are it's arguments